tests/test_s3_specific.py

```diff
@@ -129,6 +129,38 @@ def _execute_on_subprocess_and_observe(use_threads):
         assert _execute_on_subprocess_and_observe(use_threads=True) > 10
 
 
+def test_fake_directories(live_s3_like_rig):
+    """S3 can have "fake" directories created
+    either in the AWS S3 Console or by uploading
+    a 0 size object ending in a `/`. If these objects
+    exist, we want to treat them as directories.
+
+    Note: Our normal tests do _not_ create folders in this
+    way, so this test is the only one to exercise these "fake" dirs.
+
+    Ref: https://github.com/boto/boto3/issues/377
+    """
+    if live_s3_like_rig.live_server:
```

Comment:
The existing tests used this format, so I cleaned up all of them to do `if not live_s3_like_rig.live_server: pytest.skip()`.

I think this is best of both worlds since now we could have different env vars per rig/backend at some point if we wanted without any changes at the test level, but we still show skip rather than pass if we don't run it.